essence-of-live-coding/src/LiveCoding/Migrate/NoMigration.hs

```diff
@@ -0,0 +1,92 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE Arrows #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE DeriveTraversable #-}
+
+{-|
+Module      : LiveCoding.Migrate.NoMigration
+Description : Mechanism to save state in a Cell without requiring a Data instance.
+
+If a data type is wrapped in 'NoMigration' then it can be used in a 'Cell' without requiring 
+it to have a 'Data' instance. The consequence is that if the type has changed
+in between a livereload, then the previous saved value will be discarded, and no migration attempt
+will happen.
+
+'LiveCoding' does not export 'delay' and 'changes' from this module. These functions should be
+used with a qualified import.
+-}
+module LiveCoding.Migrate.NoMigration where
+
+-- base
+import Data.Data
+    ( Data(dataTypeOf, gunfold, toConstr),
+      Typeable,
+      mkConstr,
+      mkDataType,
+      Constr,
+      DataType,
+      Fixity(Prefix) )
+
+-- essence-of-live-coding
+import LiveCoding.Cell ( Cell(Cell) )
+import LiveCoding.Cell.Monad ( hoistCellKleisli )
+import Control.Arrow ( Arrow(arr), returnA, (>>>) )
+import qualified LiveCoding.Cell.Feedback as Feedback
+import Control.Monad (guard)
+
+-- |Isomorphic to @'Maybe' a@ but has a different Data instance
+data NoMigration a = Initialized a | Uninitialized
+    deriving (Show, Eq, Functor, Foldable, Traversable)
+
+fromNoMigration :: a -> NoMigration a -> a
+fromNoMigration _ (Initialized a) = a
+fromNoMigration a Uninitialized = a
+
+dataTypeNoMigration :: DataType
+dataTypeNoMigration = mkDataType "NoMigration" [initializedConstr, uninitializedConstr]
+
+initializedConstr :: Constr
+initializedConstr = mkConstr dataTypeNoMigration "Initialized" [] Prefix
+
+uninitializedConstr :: Constr
+uninitializedConstr = mkConstr dataTypeNoMigration "Uninitialized" [] Prefix
+
+-- |This 'Data' instance is such that deserialization always returns an Uninitialized value.
+instance (Typeable a) => Data (NoMigration a) where
+  dataTypeOf _ = dataTypeNoMigration
+  toConstr (Initialized _) = initializedConstr
+  toConstr Uninitialized = uninitializedConstr
+  gunfold _cons nil _ = nil Uninitialized
+
```

Comment:
It would be great to have 2 Haddock section headers, "Data instance", and "Utility functions" (further down)